From cc0e2fd24f623c617ae2c171105ae2924c818007 Mon Sep 17 00:00:00 2001
From: Zhipeng Xie <xiezhipeng1@huawei.com>
Date: Sun, 14 Nov 2021 15:57:55 +0800
Subject: [PATCH 02/37] create-diff-object: fix symbol changed sections error
 on aarch64

$d is reserved symbols in aarch64, we met following error when
building patch on aarch64. just ignore it.
ERROR: cmdline.o: symbol changed sections: $d

Signed-off-by: Zhipeng Xie <xiezhipeng1@huawei.com>
---
 kpatch-build/create-diff-object.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/kpatch-build/create-diff-object.c b/kpatch-build/create-diff-object.c
index 0b0b06b..3b677d3 100644
--- a/kpatch-build/create-diff-object.c
+++ b/kpatch-build/create-diff-object.c
@@ -892,6 +892,8 @@ static void kpatch_compare_correlated_symbol(struct symbol *sym)
 		if ((sym2->sec->twin && sym2->sec->twin->ignore) ||
 		    kpatch_subsection_changed(sym1->sec, sym2->sec))
 			sym->status = CHANGED;
+		else if (sym1->name[0] == '$') /* reserved symbols in aarch64 */
+			log_debug("maping symbols: %s", sym1->name); /* do nothing just ignogre */
 		else
 			DIFF_FATAL("symbol changed sections: %s", sym1->name);
 	}
-- 
2.33.0

